### PR TITLE
CA-31 Updates to make EMR Actions use ABAC - Tag Conditions inside IAM Policies

### PR DESCRIPTION
**Pull request by hbarros-caylent**
This is still WIP. There are a few tests left to be done.
- Adds var `emr_abac_valid_tags` into main module.
- Adds condition inside policy using the new map var `emr_abac_valid_tags`
- Refactors arn string
- Refactors few statements to gather more actions together

Tests:
- [x] [static-hbase] Fail running job when Policy has tag conditions and app is not configured to AddTags
- [x] [static-hbase] Successfully runs job when Policy has tag conditions and app is configured to AddTags
- [x] [ephemeral-spark] Fail running job when Policy has tag conditions and app is not configured to AddTags
- [x] [ephemeral-spark] Successfully runs job when Policy has tag conditions and app is configured to AddTags
- [x] It should not fail in case we don't create Managed SG and AWS creates automatically with default tag.

This PR links to [CA-29](https://github.com/Datatamer/terraform-aws-emr/pull/52) and [CA-33](https://github.com/Datatamer/terraform-aws-tamr-config/pull/15)  and they all should be merged so that Epic CA-28 is done.

### CODE REVIEW

File: modules/aws-iam-policies/main.tf [hunk 1]
```diff
@@ -1,3 +1,7 @@
+locals {
+  arn_prefix_elasticmapreduce = "arn:${var.arn_partition}:elasticmapreduce:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}"
```
Comment (souza-dan) [inline modules/aws-iam-policies/main.tf:2]: Nice, this makes the rest much more readable!

**Review by souza-dan — COMMENTED** [review]
Changes look solid! Seems like this would fit within a minor version bump so we should update version and the changelog.

File: variables.tf [hunk 2]
```diff
@@ -123,3 +123,9 @@ variable "tags" {
   description = "A map of tags to add to all resources."
   default     = {}
 }
+
+variable "emr_abac_valid_tags" {
+  type        = map(list(string))
+  description = "A map of valid tags for maintaining EMR resources when using ABAC IAM Policies with Tag Conditions. Make sure your tamr-config.yml file specifies tags key values. Refer to tamr-config module examples for more info."
```
Comment (souza-dan) [inline variables.tf:129]: The instructions about the tamr configuration is a bit vague. We should either find a way to make this guidance more specific, or we leave it completely up to the configuration module to guide the user on how to use this field.

I'm leaning toward the latter, but open to either.

**Review by souza-dan — COMMENTED** [review]
Change looks solid! We should update the changelog and minor version.

File: variables.tf (hunk 2 above)
Comment (hbarros-caylent) [reply variables.tf:129]: Added examples into configuration module https://github.com/Datatamer/terraform-aws-tamr-config/pull/15

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # Tamr VM Terraform Module
 
+## v3.3.0 - August 4th 2021
+* Adds new variable `emr_abac_valid_tags` to be used in IAM Policies conditions for creating EMR Resources using ABAC
```
Comment (souza-dan) [inline CHANGELOG.md:4]: Does upgrading from v3.2.0 to this go smoothly since it's just an update to the policy?

**Review by souza-dan — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 3 above)
Comment (hbarros-caylent) [reply CHANGELOG.md:4]: Yes! Terraform definitely won't destroy anything, only updates in-place. Older statements were either kept or merged and we shouldn't expect any problems regarding new permissions as well.

**Pull request merged**